binsync/decompilers/ghidra/client/src/main/java/binsync/BSGhidraServerAPI.java

```diff
@@ -471,15 +475,36 @@ public Map<String, Map<String, Object>> getFunctions() {
 		Map<String, Map<String, Object>> funcs = new HashMap<>();
 		for (Function func: fm.getFunctions(true)) {
 			Map<String, Object> func_data = new HashMap<>();
+			Map<Integer, Map<String, Object>> stack_vars = this.getStackVariables("0x"+func.getEntryPoint().toString(false, 0));
 			String addr = "0x"+func.getEntryPoint().toString(false, 0);
 			String name = func.getName();
 			int size = (int) func.getBody().getNumAddresses();
 			func_data.put("name", name);
 			func_data.put("size", size);
+			func_data.put("stack_vars", stack_vars);
 			funcs.put(addr, func_data);
 		}
 		
 		return funcs;
 	}
 	
+	public Map<Integer, Map<String, Object>> getStackVariables(String addr) {
+		var program = this.server.plugin.getCurrentProgram();
+		var func = this.getNearestFunction(this.strToAddr(addr));
+		Map<Integer, Map<String, Object>> stackVars = new HashMap<>();
+		for (Variable v : func.getAllVariables()) {
```

Comment:
How confident are you that this is the right way to get the stack variables? I seem to have used it different before: https://github.com/mahaloz/decomp2dbg/blob/69ad8ede239cdb5eab8a776f33426b980277cd2c/decompilers/d2d_ghidra/src/main/java/decomp2dbg/D2DPlugin.java#L201

Reply:
It works for the names. The type does change but it is incorrect. I changed a type from undefined to int, and it put undefined8 in the .toml. This is because of the toString() function which I think you mentioned earlier. All the other information seems to be correct, but I can replicate the d2d code if you want

Reply:
Considering we need the type for later things, let's replicate the d2d code. Note: doing it this way will require you to decompile the function. It's smart to just pass the decompiled function around if you do it once earlier already.

Reply:
The types get passed correctly to the project now